### PR TITLE
NF: listener's onProgressUpdate take a single element

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Currently, it takes an array, and we never use it, we always use it's first element. It's even more confusing because
this first element is a task data which contains any type. Sometime it contains an array of object. I strongly believe
it will make update slightly less confusing. At least for the few time when we actually use it

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 1]
```diff
@@ -573,8 +573,8 @@ public void onPreExecute() { /* do nothing */}
 
 
         @Override
-        public void onProgressUpdate(TaskData... values) {
-            displayNext(values[0].getCard());
+        public void onProgressUpdate(TaskData values) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:576]: `value`

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 2]
```diff
@@ -2574,8 +2574,8 @@ public void onPostExecute(TaskData result) {
 
 
         @Override
-        public void onProgressUpdate(TaskData... values) {
-            mProgressDialog.setContent(values[0].getString());
+        public void onProgressUpdate(TaskData values) {
+            mProgressDialog.setContent(values.getString());
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:2578]: value

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 3]
```diff
@@ -340,8 +340,8 @@ public void onPreExecute() {
...
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:344]: value

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 4]
```diff
@@ -223,8 +223,8 @@ public void onPreExecute() {
         }
 
         @Override
-        public void onProgressUpdate(TaskData... values) {
-            int count = values[0].getInt();
+        public void onProgressUpdate(TaskData values) {
+            int count = values.getInt();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:227]: value

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 5]
```diff
@@ -1745,7 +1745,7 @@ public void doInBackgroundLoadCollectionComplete() {
          * <p>
          * The semantics of the update data depends on the task itself.
          */
-        public void onProgressUpdate(TaskData... values) {
+        public void onProgressUpdate(TaskData values) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:1748]: value

**Comment by Arthur-Milchior** [comment]
done

**Pull request merged**